src/index.js

```diff
@@ -363,29 +432,35 @@ function setupDevice(selectedDevice, options) {
                         throw new Error(`No firmware defined for ${family} family`);
                     }
                 })
-                .then(() => validateFirmware(selectedDevice, firmwareFamily))
-                .then(valid => {
-                    if (valid) {
-                        debug('Applicaton firmware id matches');
-                        return selectedDevice;
-                    }
-                    return Promise.resolve()
-                        .then(async () => {
-                            if (!promiseConfirm) return;
-                            if (!await promiseConfirm('Device must be programmed, do you want to proceed?')) {
+                .then(() => {
+                    return confirm('Device must be programmed, do you want to proceed?')
+                        .then(proceed => {
+                            if (proceed) {
+                                return programFirmware(selectedDevice, firmwareFamily);
+                            } else {
                                 throw new Error('Preparation cancelled by user');
                             }
-                        })
-                        .then(() => programFirmware(selectedDevice, firmwareFamily));
+                        });
                 })
-                .then(() => closeJLink(selectedDevice))
-                .then(() => resolve(selectedDevice))
-                .catch(reject);
+                .then(() => closeJLink(selectedDevice), (err) => {
```

Comment:
It's hard to understand what will be the final result with this syntax, but definitely not the expected selectedDevice.